### ISSUE TITLE
headers node in body of Lambda API gateway request throws NPE 

### ISSUE THREAD

**Issue by cdthompso1** [spring-cloud/spring-cloud-function#709]
Using spring-cloud-function-adapter-aws 3.1.3-SNAPSHOT, when the payload of a request body contains a node named "headers," I receive an NPE and the following trace:
```
java.lang.NullPointerException
	at org.springframework.cloud.function.adapter.aws.AWSLambdaUtils.generateMessage(AWSLambdaUtils.java:127)
	at org.springframework.cloud.function.adapter.aws.FunctionInvoker.handleRequest(FunctionInvoker.java:80)
```
I reproduced this with the payload below using both curl  (-X POST -d "data=@./request.json") and using the JSON payload below directly in the Lambda test tool.
```
{
    "headers": {
        "X-Forwarding-Service": "ImprovMX v3.0.0",
        "Received-SPF": "pass (improvmx.com: domain of example.com designates xxx.xxx.xxx.xxx as permitted sender) receiver=mx1.improvmx.com; client-ip=xxx.xxx.xxx.xxx; helo=example.com;",
        "Delivered-To": "example@yourdomain.com",
        "DKIM-Signature": "v=1; a=rsa-sha256; c=relaxed/relaxed; d=improvmx.com; i=@improvmx.com; q=dns/txt; s=20191126; t=1581630208; h=date : from : to : subject : content-type : message-id; bh=pwCRxelulesFS4CeB7c7JkXr7yW4Mx6hmFEr6sYnDiQ=; b=hBXwNE3Akntj0cf0qC+WdATn85jDPJzyyQNHvIeOQwUVk0E9FWbcX3HPC32sXITuYbLM9 9UtVQ1GK4u/k0NMXSzJfZtjcmjV4fnmHgbmmiapdOKE4f+816p0H8Htwvj/j5+OHShLk9Iy J8KdQ3f4kfHzp46+ZRymh/Z2HUHveDU/2v2w269FYIVAdWZddtbnBB7FRHFJt29i7FKSDeR FMwuo3UBZugKY3pVMMR0sYazpgOxvBYKEv7ymhtIl/nxcxq6xb2vc0qDl51HPhkwjQ5NhyT ZsDyJp/36NYaiiY4W1ZNxayxVtpJYPCCApLFZS20cfOCG3OYL8qjWZBYu1lg==",
        "Authentication-Results": "mx1.improvmx.com; spf=pass (improvmx.com: domain of example.com designates xxx.xxx.xxx.xxx as permitted sender) smtp.mailfrom=example.com; dkim=none"
    },
    "to": [
        {
            "name": "Example user",
            "email": "example@yourdomain.com"
        }
    ],
    "from": {
        "name": "Email Test",
        "email": "test@example.com"
    },
    "subject": "This is a sample email to show the webhooks",
    "message-id": "random-message-id-generated@example.com",
    "date": "Thu, 13 Feb 2020 13:44:12 -0800",
    "return-path": {
        "name": null,
        "email": "test@example.com"
    },
    "timestamp": 1581601452,
    "text": "This is a sample text that is present in the email's body as the text/plain value.",
    "html": "This is a sample text that is present in the email's body as the text/html value."
}
```

**Comment by olegz** [comment]
@cdthompso1 Can you please provide your `request.json` in its entirety so we can test?

**Comment by olegz** [comment]
Also, based on looking at the code and realizing how line 127 can trigger NPE it leads me to believe you are using v2 API gateway request which is not yet supported (see https://github.com/spring-cloud/spring-cloud-function/issues/684)

**Comment by cdthompso1** [comment]
Yes, attaching request.json.
[request.txt](https://github.com/spring-cloud/spring-cloud-function/files/6833807/request.txt)

**Comment by cdthompso1** [comment]
You're right, it's v2.

aws apigatewayv2 get-apis

{
    "Items": [
        {
            "ApiEndpoint": "https://99q3f97743.execute-api.us-east-2.amazonaws.com",
            "ApiId": "99q3f97743",
            "ApiKeySelectionExpression": "$request.header.x-api-key",
            "CreatedDate": "2021-06-14T12:41:37+00:00",
            "Description": "Created by AWS Lambda",
            "DisableExecuteApiEndpoint": false,
            "Name": "mail2papertrailFunction-API",
            "ProtocolType": "HTTP",
            "RouteSelectionExpression": "$request.method $request.path",
            "Tags": {}
        }
    ]
}

**Comment by olegz** [comment]
See #684

**Comment by olegz** [comment]
NOTE: The fix will be available in the next version of 3.1 as ell as 3.1.
Basically with 3.1.x all you need is to override lambda-event dependencies in your POM (e.g., version 3.9.0) that support v2.

With 3.2 version of s-c-function it will just work but GA release will not be available for a while.
